mds/views.py

```diff
@@ -109,38 +110,63 @@ def perform_update(self, serializer):
         serializer.save()
 
 
+class DeviceLimitOffsetPagination(LimitOffsetPagination):
+    default_limit = 500
+
+
 class DeviceViewSet(
-    mixins.CreateModelMixin,
-    mixins.RetrieveModelMixin,
-    mixins.DestroyModelMixin,
-    mixins.ListModelMixin,
-    UpdateOnlyModelMixin,
-    MultiSerializerViewSet,
+    viewsets.ModelViewSet, UpdateOnlyModelMixin, MultiSerializerViewSet
 ):
-
-    queryset = models.Device.objects.prefetch_related(
-        Prefetch(
-            "telemetries",
-            queryset=models.Telemetry.objects.filter(
-                id__in=Subquery(
-                    models.Telemetry.objects.filter(device_id=OuterRef("device_id"))
-                    .order_by("-timestamp")
-                    .values_list("id", flat=True)[:1]
+    def get_queryset(self):
+        queryset = models.Device.objects
+
+        filters = {
+            "category": "category__in",
+            "status": "telemetries__status__in",
+            "provider": "provider__id__in",
+        }
+        time_filters = {
+            "registrationDateFrom": "registration_date__gte",
+            "registrationDateTo": "registration_date__lte",
+        }
+        for (req_field, db_field) in filters.items():
+            if self.request.GET.getlist(req_field):
+                queryset = queryset.filter(
+                    **{db_field: self.request.GET.getlist(req_field)}
                 )
-            ),
-            to_attr="_latest_telemetry",
-        )
-    ).select_related("provider")
+        for (req_field, db_field) in time_filters.items():
+            if req_field in self.request.GET:
+                queryset = queryset.filter(
+                    **{db_field: dateutil.parser.parse(self.request.GET[req_field])}
+                )
+        if "id" in self.request.GET:
+            queryset = queryset.filter(id__icontains=self.request.GET["id"])
```

Comment:
this seems duplicated - consider using a single list of (db_field, request_field, fonction_to_extract_the_field_value). Note `DjangoFilterBackend` may well do all this natively already.